### PR TITLE
User Settings Page

### PR DESCRIPTION
**Pull request by andorp**

### CODE REVIEW

File: src/Action.hs [hunk 1]
```diff
@@ -76,6 +82,8 @@ instance ActionUserHandler Action where
         put $ UserLoggedIn username "session"
         persistent $ loginUser username
 
+    userState = get
+
     logout = do
         gets _username >>= persistent . logoutUser
```
Comment (np) [inline src/Action.hs]: This is not related to your commit but `gets _username` can be replaced by `use username`.

File: src/Frontend.hs [hunk 2]
```diff
@@ -208,6 +211,8 @@ type AulaTesting =
   :<|> "topics" :> GetH (Frame (PageShow [Topic]))
   :<|> "users"  :> GetH (Frame (PageShow [User]))
 
+  :<|> "user" :> "settings" :> FormH HTML (Html ()) ST
+
```
Comment (np) [inline src/Frontend.hs]: Given the new state of AulaTesting, I would avoid adding handlers we have elsewhere.

File: src/Frontend/Page/User.hs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    , profileNewPass1 :: Maybe ST
+    , profileNewPass2 :: Maybe ST
+    }
+    deriving (Eq, Show)
+
+instance FormPageView PageUserSettings where
+    type FormPageResult PageUserSettings = UserSettingData
+
+    formAction _ =
+        P.path . P.TopTesting $ P.path P.UserSettings
```
Comment (np) [inline src/Frontend/Page/User.hs]: ... and thus adjusting this to the main frontend

File: src/Frontend/Page/User.hs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            DF.form v fa $ do
+                DF.inputText "email" v >> br_ []
+                DF.inputText "old-password" v >> br_ []
+                DF.inputText "new-password1" v >> br_ []
+                DF.inputText "new-password2" v >> br_ []
+                DF.inputSubmit "ANDERUNGEN SPEICHERN"
+
+-- FIXME: Redirect to the right place
+instance RedirectOf PageUserSettings where
+    redirectOf _ = P.path $ P.TopTesting "/ideas"
```
Comment (np) [inline src/Frontend/Page/User.hs]: Likewise

File: src/Frontend/Path.hs [hunk 5]
```diff
@@ -70,6 +71,7 @@ main SpaceAll               = "space/"
 main (SpaceOne sid p)       = "space/" </> sid </> space p
 main UserAll                = "user/"
 main (UserOne (AUID uid) p) = "user/" </> cs (show uid) </> user p
+main UserSettings           = "user/settins/"
```
Comment (np) [inline src/Frontend/Path.hs]: "settins", typo?

File: src/Action.hs (hunk 1 above)
Comment (andorp) [inline src/Action.hs]: I got the following error:
````
src/Action.hs:88:13:
    Not in scope: ‘username’
    Perhaps you meant ‘_username’ (line 125)
````

Comment (Mikolaj) [inline src/Action.hs]: The lenses are defined in the same file, much below.

Comment (andorp) [inline src/Action.hs]: It just behaves strange if I put it just before the class instance.

**Comment by andorp** [comment]
I merged it accidentally. :( If there are issues, please feel free to revert.

**Pull request merged**